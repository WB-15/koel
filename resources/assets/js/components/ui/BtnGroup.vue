<template>
  <span class="btn-group">
    <slot>
      <Btn green>Foo</Btn>
      <Btn orange>Bar</Btn>
      <Btn red>Baz</Btn>
    </slot>
  </span>
</template>

<script lang="ts" setup>
import { defineAsyncComponent } from 'vue'

const Btn = defineAsyncComponent(() => import('@/components/ui/Btn.vue'))
</script>

<style lang="scss">
.btn-group {
  display: flex;
  position: relative;
  flex-wrap: nowrap;

  button {
    &:not(:first-child) {
      border-radius: 0;
    }

    &:first-of-type {
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
      border-top-left-radius: 9999px;
      border-bottom-left-radius: 9999px;
    }

    &:last-of-type {
      border-top-right-radius: 9999px;
      border-bottom-right-radius: 9999px;
    }
  }

  &[uppercased] button {
    text-transform: uppercase;
  }
}
</style>
