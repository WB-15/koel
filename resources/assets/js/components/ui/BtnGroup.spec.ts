import { expect, it } from 'vitest'
import ComponentTestCase from '@/__tests__/ComponentTestCase'
import BtnGroup from './BtnGroup.vue'
import Btn from './Btn.vue'

new class extends ComponentTestCase {
  private renderButtonToSlot (text: string) {
    return this.render(Btn, {
      slots: {
        default: text
      }
    }).html()
  }

  protected test () {
    it('renders', () => {
      expect(this.render(BtnGroup, {
        slots: {
          default: ['Green', 'Orange', 'Blue'].map(text => this.renderButtonToSlot(text))
        }
      }).html()).toMatchSnapshot()
    })
  }
}
