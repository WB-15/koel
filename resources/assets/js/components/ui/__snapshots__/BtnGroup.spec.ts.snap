// Vitest Snapshot v1

exports[`renders 1`] = `<span class="btn-group" data-v-5d6fa912=""><button type="button" data-v-27deb898="">Green</button><button type="button" data-v-27deb898="">Orange</button><button type="button" data-v-27deb898="">Blue</button></span>`;
